src/_posts/platform/app/task-scheduling/2000-01-01-custom-clock-processes.md

```diff
@@ -1,55 +1,118 @@
 ---
 title: Scheduled Jobs with Custom Clock Processes
 nav: Custom Clock Processes
-modified_at: 2021-10-19 10:00:00
+modified_at: 2023-02-01 17:00:00
 tags: task-scheduling
 index: 2
 ---
 
-With custom clock processes, you have the ability to specify custom schedule. You will also have other benefits like
-environment parity between development and production, or time precision in your task scheduling.
+Scalingo provides a feature called *Custom Clock Processes* to help you run
+tasks on a regular basis. Unlike jobs setup through the [Scalingo Scheduler]({% post_url platform/app/task-scheduling/2000-01-01-scalingo-scheduler.md %}),
+custom clock processes do not suffer from many limitations.
 
-## Definition of the Custom Clock Process
+## About Custom Clock Processes
 
-The implementation of the custom clock will vary depending on the used language. However, defining the clock process is standard and is done in the `Procfile`.
+Custom clock processes allow to define tasks so that they run periodically.
+They give you full control over the schedule, periodicity and conditions on
+which the jobs are launched, making them a powerful alternative to the Scalingo
+Scheduler.
 
-### Example
+A custom clock process is basically a background process running indefinitely
+in a container and waking up at specified interval(s) to launch some job(s).
+This background process is responsible for controlling what job(s) must be
+launched and when. It really acts as a job scheduler.
 
-Here is an example of the process definition using the Ruby library `clockwork`:
+It's usually written in the language of your choice. This means that your
+custom clock process can either share the same code base as your application's
+or a dedicated one.
 
-```yaml
-web: bundle exec puma -C config/puma.rb
-clock: bundle exec clockwork clock.rb
-```
+Custom clock processes also work with [Review Apps]({% post_url platform/app/2000-01-01-review-apps.md %}),
+just like the parent application does.
 
-## Start the Task Scheduler
+### Limitations
 
-Once your application has been deployed, scale your 'clock' to 1 to start the task
-scheduler:
+Custom clock processes have very few limitations:
+- For deployments: [deployment limits]({% post_url platform/deployment/2000-01-01-limits.md %})
+- For containers: [containers limits]({% post_url platform/internals/2001-01-01-container-sized.md#container-limits %})
 
-```bash
-$ scalingo --app my-app scale clock:1
+They **do not** have the limitations imposed by the Scalingo Scheduler.
+Consequently, they allow to:
+
+- Run (very) long process. The jobs you run in your custom clock process
+  container(s) don't have a limited lifespan.
+
+- Run as many scheduled jobs as you want.
+
+- Have full control over the schedule: if you want a task to repeat every
+  minute, you can.
+
+- Have a very precise schedule, if need be.
+
+### Costs
+
+The feature itself it completely free, but the container in which the
```

Comment:
I don't really like calling it "the feature", it's not present in the dashboard nowhere, it's not "free". Or I would say "defining a new type of container for your custom clock is free, but the container [...]"

Reply:
I updated the paragraph in https://github.com/Scalingo/documentation/pull/1935/commits/a7a03d02828f4da629fa74557a6f3ed9c866f9c6
Please tell me if it's better?